internal/performance/create_job_test.go

```diff
@@ -0,0 +1,59 @@
+package performance
```

Comment:
Splitting out the tests into separate functions makes a lot of sense to me. Splitting out the functions into separate files seems fine, but not necessarily idiomatic in the sense that you're no longer [grouping related code together](https://medium.com/@benbjohnson/structuring-applications-in-go-3b04be4ff091#982a). Not a big deal, just a suggestion.